server/importer.js

```diff
@@ -1429,6 +1474,140 @@ var async = require('async'),
+									imported++;
+									Importer.progress(count, total);
+									return done();
+								}
+
+								if (err) {
+									Importer.warn('skipping vote:_vid: ' + _vid + ' : ' + err);
+									Importer.progress(count, total);
+									return done();
+								}
+
+								Importer.log('[process-count-at:' + count + '] saving vote:_vid: ' + _vid);
+
+								async.parallel([
+										function(cb) {
+											Data.getImportedPost(vote._pid, function(err, post) {
+												if (err) {
+													Importer.warn('getImportedPost: ' + vote._pid + ' err: ' + err);
+												}
+												cb(null, post);
+											});
+										},
+										function(cb) {
+											Data.getImportedTopic(vote._tid, function(err, topic) {
+												if (err) {
+													Importer.warn('getImportedTopic: ' + vote._tid + ' err: ' + err);
+												}
+												cb(null, topic);
+											});
+										},
+										function(cb) {
+											Data.getImportedUser(vote._uid, function(err, user) {
+												if (err) {
+													Importer.warn('getImportedTopic: ' + vote._uid + ' err: ' + err);
+												}
+												cb(null, user);
+											});
+										}
+									],
+									function(err, results){
+										var post = results[0];
+										var topic = results[1];
+										var user = results[2] || {uid: '0'};
+
+										if (!post && !topic) {
+											Importer.warn('[process-count-at: ' + count + '] post and topic do not exist! Likely it was deleted. _vid: ' + _vid);
+											done();
+										} else {
+
+											var onCreate = function(err, voteReturn) {
+												if (err) {
+													Importer.warn('skipping vote:_vid: ' + _vid + ' : ' + err);
+													Importer.warn(post);
+													Importer.warn(topic);
+													Importer.warn(user);
+													Importer.progress(count, total);
+													return done();
+												}
+
+												Importer.progress(count, total);
+
+												vote.imported = true;
+												imported++;
+												vote = nodeExtend(true, {}, vote, voteReturn);
+												votes[_vid] = vote;
+												console.log ('!!! setting vote as imported: ', _vid);
+												Data.setVoteImported(_vid, vote._action, vote, done);
+											};
+
+											var sendVote = function(pid, uid, action) {
+												console.log('!!! sending vote for ', pid, uid, action);
+												if (action == 'down') {
+													Favourites.downvote(pid, uid, onCreate);	
+												} else {
+													Favourites.upvote(pid, uid, onCreate);
+												}
+											};
+											var pid;
+											if (!_.isUndefined(post) && !_.isNull(post)) {
+												pid = post.pid;
+												console.log ('!!! pid from post is', pid);
+											} else if (!_.isUndefined(topic) && !_.isNull(topic)) {
+												pid = topic.tid;
+												console.log ('!!! pid from topic is', pid);
+											}
+
+											var action = vote._action == 2 ? 'down' : 'up';
```

Comment:
Yeah that was a carryover since I was only considering votes import from Vanilla. I've [updated my plugin](https://github.com/bdharrington7/nodebb-plugin-import-vanilla/commit/23863c3ce14179706ff99d5f85ff5cd51787cf7e#diff-168726dbe96b3ce427e7fedce31bb0bcR338) so that breaking change won't break mine :)